services/app-api/src/authn/thirdPartyAuthn.ts

```diff
@@ -0,0 +1,31 @@
+import type { Result } from 'neverthrow'
+import { ok, err } from 'neverthrow'
+import type { UserType } from '../domain-models/index'
+import type { Store } from '../postgres'
+import { lookupUserAurora } from './cognitoAuthn'
+
+export async function userFromThirdPartyAuthorizer(
+    store: Store,
+    userId: string
+): Promise<Result<UserType, Error | undefined>> {
+    try {
+        // Lookup user from postgress
+        const auroraUser = await lookupUserAurora(store, userId)
+        if (auroraUser instanceof Error) {
+            return err(auroraUser)
+        }
+
+        if (auroraUser instanceof Error) {
+            return err(auroraUser)
+        }
+
+        if (auroraUser === undefined) {
```

Comment:
@macrael if a user isn't stored in postgres we won't be able to use cognito to create and insert a new user into postgres like we're doing in `cognitoAuthn`, but I'm wondering if there's anything we should do or if erroring is okay?

Reply:
Let's just error. Like you said we don't have the info to be able to create the user. This should be pretty impossible since someone will have had to login with cognito to generate a JWT so the user better be in the DB already.